Spec Authority set Id and Gossip Message

lamafab opened the pull request:
PR for issue #195. This also defines the "Gossip Message" type.

Also, please clarify "Finalizing Message" -> is this the `PrimaryPropose` in [this type](https://docs.rs/finality-grandpa/0.12.3/src/finality_grandpa/lib.rs.html#224-238)?

```rust
/// A protocol message or vote.
#[derive(Clone, PartialEq, Eq)]
#[cfg_attr(any(feature = "std", test), derive(Debug))]
#[cfg_attr(feature = "derive-codec", derive(Encode, Decode))]
pub enum Message<H, N> {
	/// A prevote message.
	#[cfg_attr(feature = "derive-codec", codec(index = "0"))]
	Prevote(Prevote<H, N>),
	/// A precommit message.
	#[cfg_attr(feature = "derive-codec", codec(index = "1"))]
	Precommit(Precommit<H, N>),
	/// A primary proposal message.
	#[cfg_attr(feature = "derive-codec", codec(index = "2"))]
	PrimaryPropose(PrimaryPropose<H, N>),
}
```

Printscreens:
![image](https://user-images.githubusercontent.com/42901763/82492744-fca44100-9ae6-11ea-86f0-4d3d43f637a5.png)

![image](https://user-images.githubusercontent.com/42901763/82504303-51ea4d80-9afb-11ea-8bf3-69d9a3eee140.png)

![image](https://user-images.githubusercontent.com/42901763/82503795-392d6800-9afa-11ea-9a62-9896e30715e7.png)

lamafab committed:
redefine vote message formula

lamafab committed:
define authority set id

lamafab committed:
define and gossip message and reference it

lamafab committed:
reference variant section

lamafab committed:
typo

drskalman commented on polkadot-host-spec/c06-consensus.tm:
```diff
@@ -997,6 +1005,24 @@
   <subsection|GRANDPA Messages Specification>
 
+  <\definition>
+    <label|defn-gossip-message><strong|Gossip Message> is a variant as
```
Is this Grandpa specific, i.e. we only have a Gossip Message for Granpa then better calling it GRANDPA gossip. If it is used somewhere else then better define it in the network section.

lamafab replied:
Done!

drskalman committed:
- [X] Better definition for the estimate/best-final-candidate.
 - [X] Define V_obs,pot
 - [X] add 'Primary should propose' to grandpa play
 - [X] Prevote to BASE-to-GHOST
 - [X] back to tmbook style because @FlorianFranzen likes center.

drskalman committed:
- Best-Prevote-candidate is the GHOST.
- Example of non-finalized final candidate added.

drskalman committed:
de-typo-tization

lamafab committed:
rename Gossip Message to GRANDPA Gossip

lamafab committed:
mention signed/unsigned for authority set id
